chat.go

```diff
@@ -51,9 +52,32 @@ type PromptAnnotation struct {
 	ContentFilterResults ContentFilterResults `json:"content_filter_results,omitempty"`
 }
 
+const (
+	ImageURLDetailHigh = "high"
+	ImageURLDetailLow  = "low"
+	ImageURLDetailAuto = "auto"
+)
+
+type ChatMessageImageURL struct {
+	URL    string `json:"url,omitempty"`
+	Detail string `json:"detail,omitempty"`
```

Comment:
Could we please make it more type-safe, by using a separate type for `Detail`?

Something like

```go
type ImageURLDetail string

const (
	ImageURLDetailHigh ImageURLDetail = "high"
	ImageURLDetailLow  ImageURLDetail = "low"
	ImageURLDetailAuto ImageURLDetail = "auto"
)

type ChatMessageImageURL struct {
	URL    string `json:"url,omitempty"`
	Detail ImageURLDetail `json:"detail,omitempty"`
}
```

Reply:
I did this, even that I don't really like the idea. Could you explain why this is useful here and not with message roles?

Reply:
@rkintzi it adds type safety, so the compiler can do more checks for us. I think we should do this for roles too!